packages/react/src/components/search/search.react.tsx

```diff
@@ -0,0 +1,25 @@
+import { c, classy } from '@onfido/castor';
+import { Icon } from '@onfido/castor-react';
+import React from 'react';
+import { withRef } from '../../utils';
+import { Input, InputProps } from '../input/input.react';
+
+/**
+ * @note `Search` uses an `Icon` that requires `Icons` (SVG sprite) to be
+ * included in your app.
+ *
+ * https://github.com/onfido/castor-icons#use-with-plain-code
+ */
+export const Search = withRef(
+  (
+    { className, style, ...restProps }: SearchProps,
+    ref: SearchProps['ref']
+  ) => (
+    <div className={classy(c('search'), className)} style={style}>
+      <Input {...restProps} ref={ref} type="search" />
+      <Icon name="search" />
+    </div>
+  )
+);
+
+export type SearchProps = Omit<InputProps, 'children' | 'invalid' | 'type'>;
```

Comment:
Do you need `children` here?

Reply:
Not anymore, since we're not including the label in `Search` anymore, so I omit `children`.
i.e.

```ts
<label>
  Foo bar
  <Search />
</label>
```

Reply:
I meant, do you need to omit it here? Does `InputProps` even have children?

Reply:
It does, but nice catch, we should Omit there instead

Reply:
Let me know how it looks now https://github.com/onfido/castor/pull/109/commits/8069044808b06209c1bbd0782cda76f20de0451e

Reply:
All good, just maybe `eslint-disable @typescript-eslint/no-explicit-any` could be on that one line only? Otherwise happy to be merged.

Reply:
We need that in two lines.
Which of these do you prefer?

```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const withRef = <C extends FC<any>>(component: C): C =>
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  forwardRef(component as any) as any;
```

```ts
/* eslint-disable @typescript-eslint/no-explicit-any */
export const withRef = <C extends FC<any>>(component: C): C =>
  forwardRef(component as any) as any;
/* eslint-enable @typescript-eslint/no-explicit-any */
```

```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const withRef = <C extends FC<any>>(component: C): C =>
  forwardRef(component as never) as never;
```

Reply:
I do prefer 3rd option, however not sure I understand `never`? Is it better here than using `any`?

If it's not, then... Probably I would stick with 2nd option, just to have this bounded in a content insead of applied to the whole file.

Reply:
They both achieve the same thing, though I'd argue `any` is clearer than `never`, the latter is misleading.

Reply:
So you choose whiever you think is best!